database/factories/ModelFactory.php

```diff
@@ -48,55 +61,67 @@
             ])->id;
         },
         'northstar_id' => $this->faker->northstar_id,
-        'url' => $url,
         'text' => $faker->sentence(),
         'location' => 'US-'.$faker->stateAbbr(),
-        // @TODO: Only set school if the action is set to collect school ID.
-        'school_id' => $this->faker->school_id,
+        /**
+         * Although our action models are set to collect school, not all users will create posts
+         * on behalf of their school (because they don't have a school_id set on their
+         * Northstar profile).
+         */
+        'school_id' => rand(0, 1) ? $this->faker->school_id : null,
```

Comment:
You could use Faker's [`optional` modifier](https://github.com/fzaninotto/Faker#modifiers) to make this a little more readable:

```
'school_id' => $this->faker->optional()->school_id,
```